k6-operator: polish troubleshooting webpage (#1978)
grafana/k6-docs: docs/sources/k6/next/shared/k6-operator/troubleshooting-common-scenarios.md

If there's at least one runner Pod that OOM-ed, the whole test will be [stuck](https://github.com/grafana/k6-operator/issues/251) and will have to be deleted manually:

```bash
kubectl delete testrun my-test
```

A `PrivateLoadZone` test or a `TestRun` [with cloud output](https://grafana.com/docs/k6/latest/set-up/set-up-distributed-k6/usage/k6-operator-to-gck6/#cloud-output) will be aborted by the Grafana Cloud k6 after its expected duration is up.

#### How to fix

In the case of OOM, it makes sense to review the k6 script to understand what kind of resource usage this script requires. It may be that the k6 script can be improved to be more performant. Then, set the `spec.runner.resources` in the `TestRun` CRD, or `spec.resources` in the `PrivateLoadZone` CRD accordingly.

### Disruption of the k6 runners

> The k6 test can execute for a long time. But depending on the Kubernetes setup, it may happen that the Pods running k6 are disrupted and moved elsewhere during execution. This will skew the result of the test. In the case of a `PrivateLoadZone` test or a `TestRun` [with cloud output](https://grafana.com/docs/k6/latest/set-up/set-up-distributed-k6/usage/k6-operator-to-gck6/#cloud-output), the test run may additionally be aborted by Grafana Cloud k6 once its expected duration is up, regardless of the exact state of k6 processes.

Reviewer: ```suggestion
A k6 test can be executed for a long time. But depending on the Kubernetes setup, it's possible that the Pods running k6 are disrupted and moved elsewhere during execution. This will skew the test results. In the case of a `PrivateLoadZone` test or a `TestRun` [with cloud output](https://grafana.com/docs/k6/latest/set-up/set-up-distributed-k6/usage/k6-operator-to-gck6/#cloud-output), the test run may additionally be aborted by Grafana Cloud k6 once its expected duration is up, regardless of the exact state of k6 processes.
```

#### How to fix

> Ensure that k6 Pods cannot be disrupted by the Kubernetes setup, e.g. with [PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) and less aggressive configuration of autoscaler.

Reviewer: ```suggestion
Ensure that k6 Pods can't be disrupted by the Kubernetes setup, for example, with [PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) and a less aggressive configuration of the autoscaler.
```
